CHANGES.rst

```diff
@@ -4,6 +4,40 @@ Changes
 `Unreleased <https://github.com/crim-ca/weaver/tree/master>`_ (latest)
 ========================================================================
 
+Changes:
+--------
+
+- Add ``weaver.wps_workdir`` configuration setting to define the location where the underlying ``cwltool`` application
+  should be executed under. This can allow more control over the scope of the mounted volumes for *Application Package*
+  running a docker image.
+- Add mapping of WPS results from the ``Job``'s UUID to generated `PyWPS` UUID for outputs, status and log locations.
+- Add *experimental* configuration settings ``weaver.cwl_euid`` and ``weaver.cwl_egid`` to provide effective user/group
+  identifiers to employ when running the CWL *Application Package*. Using these require good control of the directory
+  and process I/O locations as invalid permissions could break a previously working job execution.
+- Add more logging configuration and apply them to ``cwltool`` before execution of *Application Package*.
+- Enforce ``no_match_user=False`` and ``no_read_only=False`` of ``cwltool``'s ``RuntimeContext`` to ensure that docker
```

Comment:
Both `no_match_user` and `no_read_only` are now *enforced* using the above values, so input files are readonly and user is mapped. Applications that expect non-readonly files can use the staging method using `InitialWorkDirRequirement` which is the standard CWL way of doing it. This makes the file to be loaded in the runtime dir, where permissions are writable to allow dumping the process outputs.
 
Only issue that I notice while testing; since `weaver` is executed with docker-compose with admin-level user, the mapped user is `0:0`. So unless we override the user ids with some 'worker' to run everything, the apps are still executed as root...

Reply:
So since our applications expect non-readonly files, you confirm that with `InitialWorkDirRequirement` they are still working?
For the root issue, that's why you introduced the user id, group id settings? Maybe you could use a string and use the gpt-init script trick to extract their values at runtime?

Reply:
Got a call, everything is fine! Will postpone the user/group handling because the solution should be to run weaver as a limited user rather than root.